### PR TITLE
JAMES-3617 Add prometheus-grafana-dashboards

### PR DESCRIPTION
**Pull request by fiery06**

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
Is docs/modules/servers/pages/distributed/operate/metrics.adoc up to date?

Can you share some screenshot of Prometheus graphs?

**Comment by chibenwa** [comment]
src/site/xdoc/server/metrics.xml can have some dead links too...

File: docs/modules/servers/pages/distributed/operate/metrics.adoc [hunk 1]
```diff
@@ -125,9 +129,13 @@ SMTP latencies reports
 - https://github.com/apache/james-project/tree/master/server/grafana-reporting/prometheus-datasource/James_SMTP_CountBoard.json[SMTP count] :
 Request count for the SMTP protocol
 
-This is for instance how the JMAP dashboard looks like: +
-
-image::metrics.png[metrics for the JMAP protocol request latencies]
+Some example dashboards: +
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/metrics.adoc]: what is this `+` at the end of the line I see a bit everywhere?

Comment (fiery06) [reply docs/modules/servers/pages/distributed/operate/metrics.adoc]: Enter for new line in .adoc format

Comment (Arsnael) [reply docs/modules/servers/pages/distributed/operate/metrics.adoc]: Between adoc and markdown I keep getting mixed up ahah ok :)

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Can we also have the PNG uploaded as comments of the PR?

**Comment by fiery06** [comment]
> Can we also have the PNG uploaded as comments of the PR?

Yes, I already added some PNG in `docs/modules/servers/assets/images/`
Tried to fix the metrics of course, to get more pictures. But only JVM, JMAP, IMAP have something enough for sampling.

**Comment by chibenwa** [comment]
I mean

Screenshot from 2021-07-22 14-19-03

**Comment by fiery06** [comment]
Update: Metrics previews - Managed to fix more boards yesterday. ( Still issue with JMAP)
Java JVM
image
IMAP 
image
MailQueue
MailQueue
BlobStore
BlobStore

**Comment by chibenwa** [comment]
GC time => can we see the derivate?

IMAP I think the unit is "Seconds" correct?

localDelivered mails enqueue dequeue -> having the derivate would be nice

spool processing & blob store => wrong units

Do we have boards for other stuff like Mailbox Listener ? ElasticSearch indexing?

**Comment by chibenwa** [comment]
webadmin, messageFastViewHitRate, 

request, request_count executor_queuedepth, in_flight for Cassandra driver

A whole panel for significant mailets (the ones mentionned in mailetcontainer.xml and takes time...

A board for SMTP too would be nice.

**Comment by chibenwa** [comment]
BTW can you deploy your boards on pre prod?

**Comment by fiery06** [comment]
> BTW can you deploy your boards on pre prod?

Everything is in 135.125.83.218:3000.
I don't have admin rights for https://monitoring.lin-saas.dev/

**Comment by chibenwa** [comment]
Why is webadmin in the middle of the DNS dashboard?

**Comment by fiery06** [comment]
Reorganize the grafana boards on James project:

- Move /garfana-reporting to /server/garfana-reporting/elasitcsearch-datasource
- Create and provision boards for /server/garfana-reporting/prometheus-datasource 
- Update the documentation, write a README for prometheus.

Changelogs:

- Add James-webAdmin dashboard
- Update all Graph-type(old) dashboards to Timeline-type dashboards.
- Fix wrong unit when converting ES json to Prometheus json.

New isssues:

- IMAP: Missing IMAP_APPEND, IMAP_CREATE, IMAP_COPY, IMAP_MOVE
- JMAP: Missing all JMAP metrics
- DNS: Missing getAllByName,getAllByName_count
- MailQueue: No data enqueueTime:outgoing
- SMTP: Error when converting metrics data: 
image

**Pull request merged**